example/Makefile

```
@@ -0,0 +1,15 @@
+SRC := c_src/example_nif.cpp
+TARGET := $(MIX_BUILD_DIR)/example_nif.so
+
+CPPFLAGS := -std=c++17 -fvisibility=hidden -fPIC -I$(ERL_INCLUDE_DIR) -I$(FINE_INCLUDE_DIR)
+
+ifeq ($(shell uname -s),Darwin)
+    LDFLAGS := -dynamiclib -undefined dynamic_lookup
+else
+    LDFLAGS := -shared
+endif
+
+all: $(TARGET)
+
+$(TARGET): $(SRC)
+	$(CXX) $(CPPFLAGS) $(LDFLAGS) -o $@ $^
```

This is adjusted version of the Makefil from the test/ project:

```suggestion
PRIV_DIR := $(MIX_APP_PATH)/priv
NIF_PATH := $(PRIV_DIR)/libexample.so
C_SRC := $(shell pwd)/c_src

CPPFLAGS := -shared -fPIC -std=c++17 -Wall -Wextra
CPPFLAGS += -I$(ERTS_INCLUDE_DIR) -I$(FINE_INCLUDE_DIR)

ifdef DEBUG
	CPPFLAGS += -g
else
	CPPFLAGS += -O3
endif

ifndef TARGET_ABI
  TARGET_ABI := $(shell uname -s | tr '[:upper:]' '[:lower:]')
endif

ifeq ($(TARGET_ABI),darwin)
	CPPFLAGS += -undefined dynamic_lookup -flat_namespace
endif

SOURCES := $(wildcard $(C_SRC)/*.cpp)

all: $(NIF_PATH)
	@ echo > /dev/null # Dummy command to avoid the default output "Nothing to be done"

$(NIF_PATH): $(SOURCES)
	@ mkdir -p $(PRIV_DIR)
	$(CXX) $(CPPFLAGS) $(SOURCES) -o $(NIF_PATH)
```